docs/uml.py

```diff
@@ -0,0 +1,467 @@
+            for field in self.nodes[node]["fields"].values():
+                type_str = _UMLNetworkABC.model_field_str(field)
+                if field.required:
+                    cls_str += f"\t{field.alias} : {type_str}\n"
+                else:
+                    cls_str += f"\t{field.alias} : {type_str} = {field.default}\n"
+            cls_str += "}"
+
+        return cls_str
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    def _edge_to_str(self, node1: str, node2: str, index: int, detailed: bool, root_node: Optional[str] = None) -> str:
+        """
+        Parse the connection into a string for plantuml output.
+        """
+        vert_extension = "{node1} -[norank]-|> {node2} #line:gray"
+        horz_extension_lr = "{node1} --|> {node2}"
+        horz_extension_rl = "{node2} <|-- {node1}"
+        vert_association = "{node1} {card1} -[norank]-* {card2} {node2} #line:gray;text:gray : {field}"
+        horz_association_lr = "{node1} {card1} --* {card2} {node2} : {field}"
+        horz_association_rl = "{node2} {card2} *-- {card1} {node1} : {field}"
+
+        # The only purpose of this is to use less code below by using the get-method with default values
+        association: Dict[Optional[str], str] = {
+            node1: horz_association_lr,
+            node2: horz_association_rl,
+        }
+        extension: Dict[Optional[str], str] = {
+            node1: horz_extension_lr,
+            node2: horz_extension_rl,
+        }
+
+        node1_str = _UMLNetworkABC._remove_last_package(node1)
+        node2_str = _UMLNetworkABC._remove_last_package(node2)
+        if root_node == node1:
+            node1_str = "." + node1.split(".")[-1]
+        elif root_node == node2:
+            node2_str = "." + node2.split(".")[-1]
+
+        if self[node1][node2][index]["type"] == "extension":
+            return extension.get(root_node, vert_extension).format(node1=node1_str, node2=node2_str)
+        if self[node1][node2][index]["type"] == "association":
+            card1 = ""
+            card2 = ""
+            if detailed:
+                # ------ Parse the cardinality into readable strings ---------------------------------------------------
+                if self[node1][node2][index]["card1"]:
+                    if self[node1][node2][index]["card1"][0] == self[node1][node2][index]["card1"][1]:
+                        card1 = f'"{self[node1][node2][index]["card1"][0]}"'
+                    else:
+                        card1 = f'"{self[node1][node2][index]["card1"][0]}..{self[node1][node2][index]["card1"][1]}"'
+                if self[node1][node2][index]["card2"]:
+                    if self[node1][node2][index]["card2"][0] == self[node1][node2][index]["card2"][1]:
+                        card2 = f'"{self[node1][node2][index]["card2"][0]}"'
+                    else:
+                        card2 = f'"{self[node1][node2][index]["card2"][0]}..{self[node1][node2][index]["card2"][1]}"'
```

Comment:
```suggestion
                def get_cardinality_string(card_key:str)->str:
                    if self[node1][node2][index][card_key]:
                        if self[node1][node2][index][card_key][0] == self[node1][node2][index][card_key][1]:
                            return f'"{self[node1][node2][index][card_key][0]}"'
                        return f'"{self[node1][node2][index][card_key][0]}..{self[node1][node2][index][card_key][1]}"'
                    return ""
                card1 = get_cardinality_string("card1")
                card2 = get_cardinality_string("card2")
```

frage ist, ob's vom linter einen auf den deckel gibt, aber wenn tatsächlich für card1 und card2 das gleiche passiert, dann wäre es schöner, es nur einmal warten zu müssen.

Reply:
Ah ja, das kommt davon, wenn man immer erst copy pastet, und danach erst aufräumt ^^